compiler/func.go

```diff
@@ -32,7 +32,7 @@ const (
 // funcImplementation picks an appropriate func value implementation for the
 // target.
 func (c *Compiler) funcImplementation() funcValueImplementation {
-	if c.GOARCH == "wasm" {
+	if c.GOARCH == "wasm" || true {
```

Comment:
I do not understand this line of code. Seems like the `else` cannot be reached?

Reply:
Oh thank you! That's an oversight, will take a look.

Reply:
Thinking about how to fix this, because this change is necessary for this PR to work at all, because the old way (the "doubleword" implementation) cannot allow `go` on function pointers.
I see two options:
1. Using the switch implementation everywhere (basically as this PR does, but in a cleaner way). No extra work is needed, binary sizes are sometimes increased slightly (within 1%). An additional quick benefit is that some limitations in the current goroutine implementation will be lifted as well, such as not being able to use function pointers of functions that do some async operation. This would also be fixed by option 2 but that is going to take longer.
2. Implementing goroutines in a more traditional way. This requires an overhaul of the scheduler. This is a significant amount of work, and is not a portable solution: it won't work on WebAssembly (but will probably work significantly better on bare metal).

Perhaps the best way to proceed is to pick option 1 and later work towards revamping the scheduler on supported platforms.

Reply:
Agreed, option 1 is the best option at this point.

Reply:
Pinging to bring this back to top of queue, hopefully can be wrapped up soon.

Reply:
Well, the best option would indeed have been 1, but then I got excited and implemented the stack-based scheduler in option 2 anyway.

Reply:
Fixed now.